opm/simulators/flow/Transmissibility_impl.hpp

```diff
@@ -238,301 +239,240 @@ update(bool global, const TransUpdateQuantities update_quantities,
-                transBoundary_[std::make_pair(elemIdx, boundaryIsIdx)] = transBoundaryIs;
+                inside.cartElemIdx = cartMapper_.cartesianIndex(inside.elemIdx);
+                applyMultipliers_(transBoundaryIs, intersection.indexInInside(), inside.cartElemIdx, transMult);
+                transBoundary_.insert_or_assign(std::make_pair(inside.elemIdx, boundaryIsIdx), transBoundaryIs);
 
                 // for boundary intersections we also need to compute the thermal
                 // half transmissibilities
                 if (enableEnergy_ && !onlyTrans) {
-                    Scalar transBoundaryEnergyIs;
-                    computeHalfDiffusivity_(transBoundaryEnergyIs,
-                                            faceAreaNormal,
-                                            distanceVector_(faceCenterInside,
-                                                            elemIdx),
-                                            1.0);
-                    thermalHalfTransBoundary_[std::make_pair(elemIdx, boundaryIsIdx)] =
-                        transBoundaryEnergyIs;
+                    Scalar transBoundaryEnergyIs =
+                        computeHalfDiffusivity_(faceAreaNormal,
+                                                distanceVector_(inside.faceCenter, inside.elemIdx),
+                                                1.0);
+                    thermalHalfTransBoundary_.insert_or_assign(std::make_pair(inside.elemIdx, boundaryIsIdx),
+                                                               transBoundaryEnergyIs);
                 }
 
-                ++ boundaryIsIdx;
+                ++boundaryIsIdx;
                 continue;
             }
 
             if (!intersection.neighbor()) {
                 // elements can be on process boundaries, i.e. they are not on the
                 // domain boundary yet they don't have neighbors.
-                ++ boundaryIsIdx;
+                ++boundaryIsIdx;
                 continue;
             }
 
             const auto& outsideElem = intersection.outside();
-            unsigned outsideElemIdx = elemMapper.index(outsideElem);  
+            outside.elemIdx = elemMapper.index(outsideElem);
 
-            // Get the Cartesian indices of the origen cells (parent or equivalent cell on level zero), for CpGrid with LGRs.
-            // For genral grids and no LGRs, get the usual Cartesian Index.
-            unsigned insideCartElemIdx = this-> lookUpCartesianData_.template getFieldPropCartesianIdx<Grid>(elemIdx);
-            unsigned outsideCartElemIdx =  this-> lookUpCartesianData_.template getFieldPropCartesianIdx<Grid>(outsideElemIdx);
+            // Get the Cartesian indices of the origin cells (parent or equivalent cell on level zero),
+            // for CpGrid with LGRs. For general grids and no LGRs, get the usual Cartesian Index.
+            inside.cartElemIdx = this->lookUpCartesianData_.
```

Comment:
Maybe we should set all the properties of `inside` outside the intersection loop, instead of setting them over and over again for every intersection.

Reply:
it's good idea but there is some reuse of variables etc that i have not fully analyzed yet.

Reply:
Ok.

Reply:
Addressed.